### ISSUE TITLE
Issue updating thermostat

### ISSUE THREAD

**Issue by diehardguy** [nestlabs/iOS-NestDK#5]
Found a small bug on the ThermostatManager class as below.  I was getting the below error when trying to set the target temperature on my thermostat which does not have a fan.   Need to add a check if the thermostat has a Fan before setting the fanActiveTimer value.

```
- (void)saveChangesForThermostat:(Thermostat *)thermostat
{
    NSMutableDictionary *values = [[NSMutableDictionary alloc] init];
    
    [values setValue:[NSNumber numberWithInteger:thermostat.targetTemperatureF] forKey:TARGET_TEMPERATURE_F];

    //Need to check if the thermostat has a fan before setting the fanTimerActive Value
    if (thermostat.hasFan) {
        [values setValue:[NSNumber numberWithBool:thermostat.fanTimerActive] forKey:FAN_TIMER_ACTIVE];
    }
    
    [[FirebaseManager sharedManager] setValues:values forURL:[NSString stringWithFormat:@"%@/%@/", THERMOSTAT_PATH, thermostat.thermostatId]];

    //End
}
```

no-hvac-fan
error 	No hvac fan
type 	no-hvac-fan
https://developers.nest.com/documentation/cloud/error-messages#no-hvac-fan
message 	No hvac fan
HTTP status code 	400

You'll get this message if the HVAC system does not appear to have an independent fan.

**Comment by SusheelYadavAhex** [comment]
Does this SDK works in simulator . I have modified my clientID , client Api domain , Client secret key, client State and Client RedirectURL in the above SDK .  But i am getting message like (you don't have any devices  ). Note :(I have devices in my developer.nest.com). 

Currently i am running this project in Xcode 8.1 
Can you please revert back soon .

**Comment by diehardguy** [comment]
Yes worked for me. Make sure you are using the main account to access the thermostats. Sub accounts dont have api access to devices.

**Comment by SusheelYadavAhex** [comment]
Thanks for your reply ,

"Main account to access the thermostats means "  i didn't get you
. Actually from the SDK i have modified , my NestClientID , NestSeceretKey
, NestState , NestRedirectUrl . (These details i have taken from
my client ). Can you just link me the SDK what should i use in my project .
And more over can you just say me the procedure how to integrate it .


Waiting for your reply

Regards,
Susheel

**Comment by diehardguy** [comment]
Can you login to the app using the oauth flow? When you login you cannot see any devices in the structure.

Sent from my iPhone

On 2 Dec 2016, at 16:55, Susheel Kumar Yadav <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks for your reply ,

"Main account to access the thermostats means " i didn't get you
. Actually from the SDK i have modified , my NestClientID , NestSeceretKey
, NestState , NestRedirectUrl . (These details i have taken from
my client ). Can you just link me the SDK what should i use in my project .
And more over can you just say me the procedure how to integrate it .


Waiting for your reply

Regards,
Susheel

-
